File: find_all_anagrams/find_all_anagrams.go
package main

import(
	"fmt"
	"sort"
)

func sortString(s string) string {
	runes := []rune(s)
	sort.Slice(runes, func(i, j int) bool {
		return runes[i] < runes[j]
	})
	return string(runes)
}

func findAllAnagrams(s string, p string) []int {
	output := []int{}
	sortP := sortString(p)
	left := 0
	right := len(p)
	if len(p) > len(s) {
		return output
	}
	for right <= len(s) {
		str := s[left:right]
		if sortString(str) == sortP {
			output = append(output, left)
		}
		right ++
		left ++
	}
	return output
}


func main() {
	s := "cbaebabacd"
	p := "abc"
	result := findAllAnagrams(s, p)
	fmt.Println(result)
}

File: find_all_anagrams/find_all_anagrams.py
#!/usr/bin/env python3


def find_all_anagrams(s, p):
    output = []
    left = 0
    right = len(p)
    sorted_p = sorted(p)

    if len(s) < len(p):
        return output

    while right <= len(s):
        sorted_w = sorted(s[left:right])
        if sorted_w == sorted_p:
            output.append(left)

        left += 1
        right += 1

    return output


if __name__ == '__main__':
    s = "cbaebabacd"
    p = "abc"
    # Output: [0, 6]
    print(find_all_anagrams(s, p))
